lib/solidus_shipstation/api/batch_syncer.rb

```diff
@@ -54,6 +54,7 @@ def post_sync(shipstation_order, shipments)
         shipment.update_columns(
           shipstation_synced_at: Time.zone.now,
           shipstation_order_id: shipstation_order['orderId'],
+          store_id: shipstation_order['advancedOptions']['storeId']
         )
```

Comment:
I think you need to include this only if it's passed.

```suggestion
        sync_opts = {
          shipstation_synced_at: Time.zone.now,
          shipstation_order_id: shipstation_order['orderId']
        }
        store_id = shipstation_order.dig('advancedOptions', 'storeId')
        sync_opt[:store_id] = store_id if store_id
        shipment.update_columns(sync_opt)
```